src/MessageHandler/LostMessagesConsumer.php

```diff
@@ -44,33 +44,46 @@ public function setMessageTtl(int $messageTtl): void
     /**
      * Requeue message with date superior to message ttl
      * Then max retry reached, message is put in dead letter list
+     *
+     * @param int|null $secondTime
      */
-    public function requeueOldMessages()
+    public function requeueOldMessages(?int $secondTime = null): void
     {
-        array_map(
-            [$this, 'requeueOldMessageFromList'],
-            iterator_to_array($this->queue->getWorkingLists($this->redisClient))
-        );
+        array_map([$this, 'requeueOldMessageFromList'], iterator_to_array($this->queue->getWorkingLists($this->redisClient)), [$secondTime]);
     }
 
-    protected function requeueOldMessageFromList(string $list)
+    /**
+     * Requeue old messages from $list, and remove them if $messageTTL or
+     *
+     * @param string   $list
+     * @param int|null $secondTime
+     */
+    protected function requeueOldMessageFromList(string $list, ?int $secondTime = null)
     {
-        for ($i = 0, $n = $this->redisClient->llen($list); $i < $n; $i++) {
-            $serializedMessage = $this->redisClient->rpoplpush($list, $list);
+        do {
+            // If maximum times is defined and time is exceed.
+            if (($secondTime !== null && time() >= (time() + $secondTime))) {
+                break;
+            }
 
             // check if message is old enough
-            $message = MessageEnvelope::unserializeMessage($serializedMessage);
-            if ((time() - $message->getUpdatedAt()->format('U')) > $this->messageTtl
-                && $this->redisClient->lrem($list, 0, $serializedMessage)) {
-                if ($message->getRetry() < $this->maxRetry) {
-                    $this->addMessageInQueueList($message);
+            $serializedMessage = $this->redisClient->rpoplpush($list, $list);
```

Comment:
I don't understand why this is done twice (here and in the `while` statement)

Reply:
:bomb: me too. Thx for pre-review :smiley: